### ISSUE TITLE
false positive no-member with slot

### ISSUE THREAD

**Issue by fmigneault** [PyCQA/pylint#3364]
### Steps to reproduce
```python 
class SomeInterface(object):
    __slots__ = ["type"]
    def __init__(self):
        if not self.type:
            raise NotImplementedError("missing type")
```

### Current behavior
Raises message `E1101: instance of 'SomeInterface' has not 'type' member (no-member)`

### Expected behavior
Nothing raised as the member is defined by a slot.

### pylint --version output
pylint 2.4.4
astroid 2.3.3
Python 3.7.5 (default, Oct 25 2019, 15:51:11) 
[GCC 7.3.0]

**Comment by PCManticore** [comment]
Thanks for opening an issue. I don't think this is valid, as slots don't define attributes and `pylint` doesn't take a hint from slots for attribute definitions. We might use attribute assignments in the future to take a hint on what attributes might be available at runtime, as in the following example, but slots is a different thing entirely:

```
class SomeInterface(object):
    type: int
```

**Comment by fmigneault** [comment]
According to python documentation: 
https://docs.python.org/3/reference/datamodel.html?highlight=slots#slots
First line : 
`__slots__ allow us to explicitly declare data members (like properties)`
And just a bit after
`This class variable can be assigned a string, iterable, or sequence of strings with variable names used by instances. `

Given this doc, I was under the impression that using `__slots__` for defining attributes is a totally legitimate use case. 
My comprehension is that they behave similarly as "normal" attributes as they are recognized as class members, but using pre-definition to save memory space and speed-up their read/write operation. 
If they can be accessed just like a normal property to get/set a value, just as in the following example, why wouldn't pylint recognize that the attribute exists within the class?

```python
class T(object):
    __slots__ = ["type"]
    def __init__(self):
        self.type = 1
   
T().type  # prints 1 as expected
```

Also, using an attribute with the same name as an existing `__slots__` is not possible, as the class will raise a conflict error. It is therefore not possible to adjust existing code that do employ `__slots__` to fit pylint's member detection mechanism.
